src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift

```diff
@@ -0,0 +1,321 @@
+					self?.store.previousRiskExposureMetadata = self?.store.currentRiskExposureMetadata
+					completion?(result)
+				case let .failure(error):
+					Log.error("Analytics data were not submitted", log: .ppa, error: error)
+					completion?(result)
+				}
+			}
+		)
+	}
+
+	private func gatherExposureRiskMetadata() -> [SAP_Internal_Ppdd_ExposureRiskMetadata] {
+		guard let storedUsageData = store.currentRiskExposureMetadata else {
+			return []
+		}
+		return [SAP_Internal_Ppdd_ExposureRiskMetadata.with {
+			$0.riskLevel = convertToProto(storedUsageData.riskLevel)
+			$0.riskLevelChangedComparedToPreviousSubmission = storedUsageData.riskLevelChangedComparedToPreviousSubmission
+			$0.mostRecentDateAtRiskLevel = convertToProto(for: storedUsageData.mostRecentDateAtRiskLevel)
+			$0.dateChangedComparedToPreviousSubmission = storedUsageData.dateChangedComparedToPreviousSubmission
+		}]
+	}
+
+	// already created for EXPOSUREAPP-4790
+	/*
+	private func gatherNewExposureWindows() -> [SAP_Internal_Ppdd_PPANewExposureWindow] {
+	}
+
+	private func gatherTestResultMetadata() -> [SAP_Internal_Ppdd_PPATestResultMetadata] {
+	}
+
+	private func gatherKeySubmissionMetadata() -> [SAP_Internal_Ppdd_PPAKeySubmissionMetadata] {
+	}
+
+	private func gatherClientMetadata() -> SAP_Internal_Ppdd_PPAClientMetadataIOS {
+	}
+	*/
+
+	private func gatherUserMetadata() -> SAP_Internal_Ppdd_PPAUserMetadata {
+		guard let storedUserData = store.userMetadata else {
+			return SAP_Internal_Ppdd_PPAUserMetadata.with { _ in }
+		}
+
+		return SAP_Internal_Ppdd_PPAUserMetadata.with {
+			$0.federalState = convertToProto(storedUserData.federalState)
+			$0.administrativeUnit = Int32(storedUserData.administrativeUnit)
+			$0.ageGroup = convertToProto(storedUserData.ageGroup)
+		}
+	}
+
+	private func convertToProto(_ riskLevel: RiskLevel) -> SAP_Internal_Ppdd_PPARiskLevel {
+		switch riskLevel {
+		case .low:
+			return .riskLevelLow
+		case .high:
+			return .riskLevelHigh
+		}
+	}
+
+	private func convertToProto(for date: Date?) -> Int64 {
+		guard let date = date else {
+			return -1
+		}
+		let utcDataFormatter = ISO8601DateFormatter()
+		utcDataFormatter.formatOptions = [.withFullDate]
+
+		guard let utcDate = utcDataFormatter.date(from: utcDataFormatter.string(from: date)) else {
+			Log.warning("Trouble with converting date to utc midnight date", log: .ppa)
+			return Int64(date.timeIntervalSince1970)
+		}
+		return Int64(utcDate.timeIntervalSince1970)
+	}
+
+	private func convertToProto(_ ageGroup: AgeGroup) -> SAP_Internal_Ppdd_PPAAgeGroup {
+		switch ageGroup {
+		case .ageBelow29:
+			return .ageGroup0To29
+		case .ageBetween30And59:
+			return .ageGroup30To59
+		case .age60OrAbove:
+			return .ageGroupFrom60
+		}
+	}
+
+	// swiftlint:disable cyclomatic_complexity
+	private func convertToProto(_ federalState: FederalStateName) -> SAP_Internal_Ppdd_PPAFederalState {
```

Comment:
Can we do this as an extension on `FederalStateName `? Same for the other `convertToProto` could be something like `func protobuf() ->SAP_Internal_Ppdd_PPAFederalState`

Reply:
yay, I will do